### PR TITLE
ACM:8774: Implement federated global search

### PR DESCRIPTION
**Pull request by jlpadilla**
### Related Issue
https://issues.redhat.com/browse/ACM-8774

### Description of changes
- Initial implementation of federated global search.

### Limitations
- Does not support query aliases. https://graphql.org/learn/queries/#aliases
- HTTPS requests using `--insecure`
- LIMIT and SORT not honored in merged results.
- Relationships query not yet supported.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,12 +71,47 @@ test-scale-setup: ## Creates the search-api route in the current target cluster.
...
 
 test-send: ## Sends a graphQL query using cURL for development testing.
-	URL=https://localhost:4010/searchapi/graphql \
-	TOKEN=$(shell oc whoami -t) \
 	curl --insecure --location --request POST ${URL} \
-	--header "Authorization: Bearer ${TOKEN}" --header 'Content-Type: application/json' \
+	--header "Authorization: Bearer ${API_TOKEN}" --header 'Content-Type: application/json' \
 	--data-raw '{"query":"query q($$input: [SearchInput]) { search(input: $$input) { count items } }","variables":{"input":[{"keywords":[],"filters":[{"property":"kind","values":["ConfigMap"]}],"limit": 3}]}}' | jq
 
+# Target API URL for the test queries.
```
Comment (jlpadilla) [inline Makefile:78]: Changes here are to send test queries easily. Some examples:
```
make send QUERY=schema FEDERATED=true
make send Q=search F=true
make send Q=searchComplete
```

File: pkg/federated/mergeResponses.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		// else {
+		// 	klog.V(9).Infof("SearchComplete - Skipping duplicate value: %s", prop)
+		// }
+	}
+
+	// TODO: How to handle LIMIT ?
+	// TODO: How to handle SORT ?
+}
+
+func (d *Data) mergeSearchResults(hubName string, results []SearchResult) {
```
Comment (SherinV) [inline pkg/federated/mergeResponses.go:50]: Should hubname be an array too, if results are from different clusters?

Comment (SherinV) [reply pkg/federated/mergeResponses.go:50]: Oh, this is from one hub - not all the hubs together. Got it.

File: pkg/federated/federated.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+var getFedConfig = getFederationConfig
+var httpClientGetter = GetHttpClient
+
+func HandleFederatedRequest(w http.ResponseWriter, r *http.Request) {
+	klog.Info("Resolving federated search query.")
+	ctx := r.Context()
+	receivedBody, err := io.ReadAll(r.Body)
+	if err != nil {
+		klog.Errorf("Error reading request body: %s", err)
```
Comment (SherinV) [inline pkg/federated/federated.go:29]: Do we proceed after this err or return?

**Review by SherinV — APPROVED** [review]
/lgtm
/hold

File: pkg/federated/federated.go (hunk 3 above)
Comment (jlpadilla) [reply pkg/federated/federated.go:29]: I’ll return with error because we can’t do anything without the body.

Comment (jlpadilla) [reply pkg/federated/federated.go:29]: done

**Pull request merged**